### PR TITLE
fix(account-modal): fix cancellation tx link for Gnosis Safe

### LINKED ISSUE #2370: Link is not correct to the cancellation TX in Safe
1. Place a Market order when connected to the Safe wallet
2. Cancel the order
3. Verify the cancellation link

**AR**: it navigates to the order creation TX

**ER**: link navigates to the cancellation TX in the safe

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2370

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , great!
When I open the app inside the Safe app, I see the connect link to the cancellation TX. 
However, when I connect to the Safe using WC, I get a link to the etherscan with 404 error:
image

Could you please fix it?

Thanks!

File: src/cow-react/modules/account/containers/Transaction/StatusDetails.tsx [hunk 1]
```diff
@@ -100,7 +100,7 @@ export function StatusDetails(props: StatusDetailsProps) {
   const hasCancellationHash = !!cancellationHash && !isCancelling && !isConfirmed && isCancelled
   const cancellationTxLink = hasCancellationHash
     ? safeTransaction
-      ? getSafeWebUrl(chainId, safeTransaction.safe, safeTransaction.safeTxHash)
+      ? getSafeWebUrl(chainId, safeTransaction.safe, cancellationHash)
```
Comment (anxolin) [inline src/cow-react/modules/account/containers/Transaction/StatusDetails.tsx:103]: can this be undefined? (since you have a `?` in line 96

Comment (anxolin) [reply src/cow-react/modules/account/containers/Transaction/StatusDetails.tsx:103]: 👍 make sense

**Review by anxolin — APPROVED** [review]

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by alfetopito — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Great job!

**Pull request merged**